cmd/onos-config/onos-config.go

```diff
@@ -159,7 +165,7 @@ func main() {
 			}
 		}
 
-		mgr.Run()
+		mgr.Run(southbound.NewTarget)
```

Comment:
I know this is necessary for a target to be mocked but it completely breaks boundaries and abstractions, this is a no go for the whole patch. the target is an interface in the SB. it should never be placed in the cmd for the manager to run. we need another way to pass this in it can't come from here.

Reply:
what I would do instead is to change only the SB code with a public variable (accessibile from tests with southbound.BuildFunction or something similar) in factory that contains the TargetFn build function. that variable is set to southbound.NewTarget for production code. Before doing a .run() on the manager you swap it for `mockTargetCreationfn`
this is a solution that touches just the factory and the SB, with NO appearances in upper levels

Reply:
What about if this was passed nil and the southbound.NewTarget was taken to be the default implementation (down in Factory or synchronizer). In that way we could spare referencing southbound from here, which is what I think you objection is

Reply:
no, my objection is making the initialisation of the manager on something that should pertain only in the southbound.
what do you think of the variable solution ?